src/MPS.jl

```diff
@@ -540,20 +541,29 @@ function mixed_canonize!(tn::AbstractMPO, orthog_center)
     return tn
 end
 
+LinearAlgebra.normalize(ψ::AbstractMPO; kwargs...) = normalize!(copy(ψ); kwargs...)
+LinearAlgebra.normalize(ψ::AbstractMPO, site::Site) = normalize!(copy(ψ), site)
+
 LinearAlgebra.normalize!(ψ::AbstractMPO; kwargs...) = normalize!(form(ψ), ψ; kwargs...)
+LinearAlgebra.normalize!(ψ::AbstractMPO, site::Site) = normalize!(form(ψ), ψ; at=site)
 
 function LinearAlgebra.normalize!(::NonCanonical, ψ::AbstractMPO; at=Site(nsites(ψ) ÷ 2))
     tensor = tensors(ψ; at)
     tensor ./= norm(ψ)
     return ψ
 end
 
-LinearAlgebra.normalize!(ψ::AbstractMPO, site::Site) = normalize!(mixed_canonize!(ψ, site); at=site)
+LinearAlgebra.normalize!(config::NonCanonical, ψ::AbstractMPO; at=site) = normalize!(MixedCanonical(at), ψ; at)
```

Comment:
isn't this cheating? like you need to mixed canonize in order to use the `MixedCanonical` `normalize!` method

Reply:
you can always use the non-`Site` normalization for `NonCanonical`

Reply:
No, because inside the `normalize!(::MixedCanonical, ...)` there is already an step that does `mixed_canonize!`... I agree this is confusing. This step should be fixed in some way, since if the state is already in the `MixedCanonical` form, then it should not be necessary to do a whole `mixed_canonize!`. But for now is okay I guess.

Reply:
yes, but that `mixed_canonize!` inside the `normalize!(::MixedCanonical)` method dispatches to `mixed_canonize!(::MixedCanonical)` so it just moves the orthogonality center and doesn't call `canonize_site!` on the rest of the sites. so, if you call `mixed_canonize!` here, orthogonalizing to `at` site, then the interior `mixed_canonize!` will do nothing and that's completely fine.

so the original implementation is not more costly than this.

Reply:
Nevertheless, I noticed we had these two lines:
```julia
function LinearAlgebra.normalize!(::NonCanonical, ψ::AbstractMPO; at=Site(nsites(ψ) ÷ 2))
    tensor = tensors(ψ; at)
    tensor ./= norm(ψ)
    return ψ
end
LinearAlgebra.normalize!(ψ::AbstractMPO, site::Site) = normalize!(mixed_canonize!(ψ, site); at=site)
```
So maybe we can just use the first one, and remove the second for only `MixedCanonical` forms

Reply:
> But yeah we should do that at some point. I don't think that is a particular a priority now, right? Maybe a comment will do

yeah, leave the original implementation commented and open an issue for implemented `mixed_canonize!(::MixedCanonical)` plus the reference to this code so it's changed when this is implemented

> So maybe we can just use the first one, and remove the second for only MixedCanonical forms

mmm fine for me